.github/workflows/upload-robots-txt-file-to-release.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+name: "Upload robots.txt file to release"
+run-name: "Upload robots.txt file to release"
+
+on:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: write
+
+jobs:
+  upload-robots-txt-file-to-release:
+    name: "Upload robots.txt file to release"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v4
```

Comment:
Thanks @glyn! I've added an explicit ref to this step to make it explicitly use the tag, see e8f0784.

Although in this case the `GITHUB_REF` would have been correct, see the [release event docs], I take your point about this being subtle. If another `on:` event was added to the workflow this could subtly break so thanks for pointing that out 😄

[release event docs]: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release

Reply:
That looks better, thanks.